static/gsApp/views/amCheckout/index.tsx

```diff
@@ -643,6 +644,10 @@ class AMCheckout extends Component<Props, State> {
     const isOnSponsoredPartnerPlan =
       (subscription.partner?.isActive && subscription.isSponsored) || false;
 
+    const hasPendingChangeToFree: boolean = isDeveloperPlan(
+      subscription.pendingChanges?.planDetails
+    );
+
```

Comment:
I believe cancellations don't set the pending plan; you'd probably want to check `subscription.cancelAtPeriodEnd` and `subscription.closeAtPeriodEnd`

Reply:
When canceling from paid to free we should have pending changes because we call `change_plan` inside `cancel_plan`.

However it's possible there may not be pending changes. In `cancel_plan` we always set `cancel_at_period_end` to `True`, so using `subscription.cancelAtPeriodEnd`.

Reply:
We only call `change_plan` when `at_period_end` is not true so all regular cancellations won't have a pending plan 👍🏽